.github/workflows/ci.yml

```diff
@@ -72,6 +71,17 @@ jobs:
           cd ..
           echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/usr/lib" >> $GITHUB_ENV
           echo "TOOLCHAIN_OPTION=$TOOLCHAIN_OPTION -DCMAKE_PREFIX_PATH=$PWD/usr" >> $GITHUB_ENV
+      - name: Prepare OpenSSH Server Container for Testing
+        run: |
+          cd tests
+          set +eo pipefail
+          docker login https://docker.pkg.github.com -u libssh2 -p ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I think docker provides ready to use actions for this.

Reply:
It seems there is not one by docker, but there is one made by another user.

https://github.com/marketplace/actions/build-docker-images-using-cache

Reply:
Did you see this? https://www.docker.com/blog/docker-github-actions/

Reply:
The caching they do in that step is more-or-less identical to the caching action that this PR replaces, just done manually. They also do not perform a pull as this PR does as they rely on the cache.

Reply:
I meant using `uses: docker/login-action@v1` and `uses: docker/build-push-action@v2` instead of the `docker` commands.

Reply:
I swapped over to `docker/login-action@v1` and `docker/build-push-action@v2` from `whoan/docker-build-with-cache-action@v5`. It required adding an option to skip `docker build` in the tests themselves for whatever reason, but it works now.